### PR TITLE
nixd: 2.2.3 -> 2.3.0

### PR DESCRIPTION
**Pull request by redyf**
https://github.com/nix-community/nixd/releases/tag/2.3.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by inclyc — COMMENTED** [review]
Hi @redyf, I kindly ask do we need to add `python312` in nativeBuildInputs? Or it is already included?

Also, I very appreciate many contributions you did for this project, would you like to become a maintainer of this package?

**Comment by redyf** [comment]
> Hi @redyf, I kindly ask do we need to add `python312` in nativeBuildInputs? Or it is already included?
> 
> Also, I very appreciate many contributions you did for this project, would you like to become a maintainer of this package?

Hey inclyc, the package was built successfully without it. However I didn't notice that there was a PR adding it in the official nixd repo 5 days ago. I believe it's better to add it in this PR.
Also yes, I'd love to become a maintainer :smiley:

**Review by inclyc — APPROVED** [review]
> Hey inclyc, the package was built successfully without it. However I didn't notice that there was a PR adding it in the official nixd repo 5 days ago. I believe it's better to add it in this PR too.

Thanks, maybe just push a commit to add yourself to the maintainers list 👍 .

**Comment by redyf** [comment]
> > Hey inclyc, the package was built successfully without it. However I didn't notice that there was a PR adding it in the official nixd repo 5 days ago. I believe it's better to add it in this PR too.
> 
> Thanks, maybe just push a commit to add yourself to the maintainers list 👍 .

Done :+1:

**Comment by Aleksanaa** [comment]
Please reword the second commit to `nixd: add redyf as maintainer`

**Review by inclyc — APPROVED** [review]

**Review by inclyc — APPROVED** [review]
LGTM, why the bot requested review from me again?

**Comment by redyf** [comment]
> LGTM, why the bot requested review from me again?

I'm asking myself the same thing hahaha, ofborg what are you doing :sob:. But I think it's because most part of the evaluation was successfully completed after approving the changes.

File: pkgs/development/tools/language-servers/nixd/default.nix [hunk 1]
```diff
@@ -16,22 +16,24 @@
...
       repo = "nixd";
       rev = version;
-      hash = "sha256-1i4U012vqw84NuD5cjgfVX/he4vW9MyEeIetJ55g280=";
+      hash = "sha256-iB6Hc3kgtsZ360JlAhPqCRWFTf3tawHEKYAeYt7DM4E=";
     };
 
     nativeBuildInputs = [
       meson
       ninja
+      python312
```
Comment (drupol) [inline pkgs/development/tools/language-servers/nixd/default.nix:36]: Wouldn't it be better to use `python3` instead?

Comment (redyf) [reply pkgs/development/tools/language-servers/nixd/default.nix]: It really does, I just pushed the changes. Thanks!

**Review by drupol — APPROVED** [review]
Diff LGTM

**Review by nixf-tidy-review — APPROVED** [review]

**Pull request merged**